tests/std/tests/P0896R4_ranges_alg_for_each/test.cpp

```diff
@@ -2,74 +2,63 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <algorithm>
-#include <array>
 #include <cassert>
 #include <concepts>
 #include <functional>
 #include <ranges>
 #include <utility>
 
 #include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
 
-constexpr void smoke_test() {
-    using ranges::for_each, ranges::for_each_result, ranges::in_fun_result, ranges::iterator_t;
-    using std::identity, std::same_as;
-    using P = std::pair<int, int>;
-    using R = std::array<P, 3>;
+// Validate that for_each_result aliases in_fun_result
+STATIC_ASSERT(same_as<ranges::for_each_result<int, double>, ranges::in_fun_result<int, double>>);
 
-    // Validate that for_each_result aliases in_fun_result
-    STATIC_ASSERT(same_as<for_each_result<int, double>, in_fun_result<int, double>>);
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::for_each(borrowed<false>{}, identity{})),
+    ranges::for_each_result<ranges::dangling, identity>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::for_each(borrowed<true>{}, identity{})), ranges::for_each_result<int*, identity>>);
 
-    // Validate dangling story
-    STATIC_ASSERT(
-        same_as<decltype(for_each(borrowed<false>{}, identity{})), for_each_result<ranges::dangling, identity>>);
-    STATIC_ASSERT(same_as<decltype(for_each(borrowed<true>{}, identity{})), for_each_result<int*, identity>>);
+struct instantiator {
+    static constexpr P expected[3] = {{1, 42}, {3, 42}, {5, 42}};
 
-    R const values = {{{0, 42}, {2, 42}, {4, 42}}};
-    auto incr      = [](auto& y) { ++y; };
+    template <ranges::input_range ReadWrite>
+    static constexpr void call() {
+        using ranges::for_each, ranges::for_each_result, ranges::in_fun_result, ranges::iterator_t;
+        auto incr = [](auto& y) { ++y; };

```

Comment:
Could we please move this lambda into class or namespace scope to reduce total template instantiations?

Reply:
Done